test/unit/puppetlabs/services/jruby_pool_manager/jruby_pool_test.clj

```diff
@@ -220,53 +220,50 @@
-          (is (= first-ids second-ids third-ids fourth-ids)))
-        (let [fifth-ids (drain-and-refill pool-context 4)]
+          (is (= first-ids second-ids third-ids)))
+        (let [fourth-ids (jruby-testutils/drain-and-refill pool-context)]
           (testing "All instances flushed after max borrows"
-            (is (empty? (set/intersection fifth-ids fourth-ids))))))))
+            (is (empty? (set/intersection fourth-ids third-ids))))))))
   (testing "Splayed JRuby instance flushing -"
     (jruby-testutils/with-pool-context
      pool-context
      jruby-testutils/default-services
-      ;; with two instances, each with two max borrows, we should recycle
-      ;; an instance every cycle of draining
-     (jruby-test-config 2 2)
-     (let [first-ids (drain-and-refill pool-context 2)
-           second-ids (drain-and-refill pool-context 2)
-           old-instance-set (set/intersection first-ids second-ids)
-           new-instance-set (set/difference second-ids first-ids)]
-       (testing "Instances first flush is splayed"
-         (is (= 1 (count old-instance-set))))
-       (let [third-ids (drain-and-refill pool-context 2)]
-         (testing "Instances are not repeatedly flushed at splay interval"
-           (is (= 1 (count (set/intersection third-ids new-instance-set)))))
-         (let [fourth-ids (drain-and-refill pool-context 2)]
-           (testing "Instances are flushed at max-borrow after initial splay"
-             (is (empty? (set/intersection fourth-ids new-instance-set))))))))))
+      ;; with three instances, each with three max borrows, we should recycle
+      ;; an instance every cycle of draining. The first instance to be
+      ;; recycled will do so after the first drain, its replacement should
+      ;; then stay until the fourth drain/refill.
+     (jruby-test-config 3 3)
+     (let [first-ids (jruby-testutils/drain-and-refill pool-context)
+           second-ids (jruby-testutils/drain-and-refill pool-context)
+           third-ids (jruby-testutils/drain-and-refill pool-context)
+           fourth-ids (jruby-testutils/drain-and-refill pool-context)
+           fifth-ids (jruby-testutils/drain-and-refill pool-context)
+           original-instances-surviving-first-drain (set/intersection first-ids second-ids)
+           new-instance-after-first-drain (set/difference second-ids first-ids)
+           original-instances-surviving-second-drain (set/intersection first-ids third-ids)
+           original-instances-surviving-third-drain (set/intersection first-ids fourth-ids)]
+       (testing "Initial instances are flushed each splay interval"
+         (is (= 2 (count original-instances-surviving-first-drain)))
+         (is (= 1 (count original-instances-surviving-second-drain)))
+         (is (= 0 (count original-instances-surviving-third-drain))))
+       (testing "New instances are flushed at max-borrows"
+         (is (seq (set/intersection new-instance-after-first-drain third-ids)))
```

Comment:
`not-empty` might be more readable here than `seq`, although the docs recommend using `(seq foo)` and not `(not (empty? foo))`. they seem to ignore `(not-empty foo)`.

Reply:
Agreed on the readability part, I was applying the recommendation for `(not (empty? foo))`, but I'll go ahead and change that.